eos-tech-support/eos-fix-ostree-repo

```diff
@@ -166,29 +175,55 @@ def mark_commits_partial(repo):
             print('Commit', checksum, 'already marked as partial')
             continue
 
+        mark_partial = False
         try:
-            repo.traverse_commit(checksum, 0)
+            # If a dirtree is missing, traverse_commit will fail with
+            # G_IO_ERROR_NOT_FOUND.
+            _, reachable_objects = repo.traverse_commit(checksum, 0)
+
+            # Unfortunately, it doesn't check that the leaves (dirmeta
+            # and files) exist, so we need to do that manually. In case
```

Comment:
This sounds like something which it should do. Have you filed an upstream issue about `traverse_commit()`’s behaviour? Maybe do that and link to it in a `FIXME` comment here.

Reply:
https://github.com/ostreedev/ostree/issues/1222. Will add a comment.